packages/styled-components/src/utils/isPlainObject.js

```diff
@@ -1,2 +1,8 @@
 // @flow
-export default (x: any): boolean => typeof x === 'object' && x.constructor === Object;
+import { typeOf } from 'react-is';
+
+export default (x: any): boolean =>
+  x !== null &&
+  typeof x === 'object' &&
+  x.toString() === '[object Object]' &&
```

Comment:
Like indicated in the PR, this will need to be `Object.prototype.toString` to support objects with `toStrung`, including `Object.create(null)` 🙌

Reply:
Ah yes of course, we can't call `toString()`if it doesn't exist on that object.

The problem I was trying to address by using `x.toString()` instead of `Object.prototype.toString(x)` was to call a custom `toString` when it exists:

```js
class MyClass {
  toString() {
    return "[object MyClass]"
  }
}
const m = new MyClass();
Object.prototype.toString(m); // "[object Object]" <-- does not call the custom toString method
m.toString(); // "[object MyClass]"
```

I can solve this by using: `(x.toString ? x.toString() : Object.prototype.toString(x)) === '[object Object]'`